src/services/asyncStorageAdapter.ts

```diff
@@ -81,9 +81,20 @@ export const addPatientItem = async (key: string, newItem: Patient) => {
     console.log("⚠️  ID already exists: ", await getItem(key));
     return false;
   }
-
-  list.push(newItem);
-  await setItem(key, JSON.stringify(list));
+  let array: Patient[] = list.map(function parser(value) {
+    return (value as unknown) as Patient;
+  });
```

Comment:
Acho que dá pra fazer isso só com um type cast simples:

```suggestion
  let array = list as Patient[];
```

Mas daí tem que mudar a declaração de `array` (linha 76) para `string | any[] | null` (`any[]` tem mais sentido do que `string[]`).